hyperbolic-rs/rudders/datasets/keen.py

```diff
@@ -64,3 +66,65 @@ def build_iid2title(item_id_key, item_title_key):
             if item_id_key in line and item_title_key in line:
                 iid2title[line[item_id_key]] = line[item_title_key][1:-1]
     return iid2title
+
+
+def get_keens(data):
+    all_keens = {}
+    for item in data:
+        keen_id = item["keen_id"]
+        gem = None
+        if "gem_id" in item:
+            gem = Gem(item)
+
+        if keen_id in all_keens:
+            keen = all_keens[keen_id]
+        else:
+            keen = Keen(item["keen_id"], item["keen_title"], get_value(item, "keen_description"),
+                        item["keen_creator_uid"])
+            all_keens[keen_id] = keen
+
+        if gem is not None and not gem.is_empty():
+            keen.gems.append(gem)
+    return all_keens
+
+
+def process_input(string):
+    """
+    The json files are written in the following format:
+        {'key': '"value"'}
+
+    This function removes the initial and ending quotes and any '\n' in value.
+    Example:
+    Input: '"value1\nvalue2"'
+    Output: 'value value2'
+    """
+    string = string[1:-1] if type(string) == str else ""
+    return string.replace("\\n", " ")
```

Comment:
Optional:
- In additional to '\\n', you could replace all whitespace-type characters with a single space.
- You could add sanity checks to avoid stuff like stripping quotes that are not there or empty inputs.

Reply:
Done